main.tf

```diff
@@ -39,36 +39,32 @@ resource "aws_iam_role_policy" "event_handler_lambda_iam_role_policy" {
   name = "event_handler_lambda_iam_role_policy-${var.name}"
   role = aws_iam_role.event_handler_lambda_iam_role.id
 
-  policy = jsonencode({
-    Version = "2012-10-17"
-    Statement = [
-      {
-        "Effect" : "Allow",
-        "Action" : "lambda:InvokeFunction",
-        "Resource" : aws_lambda_function.handler_lambda.arn
-      },
-      {
-        "Effect" : "Allow",
-        "Action" : [
-          "logs:PutLogEvents",
-          "logs:CreateLogStream",
-          "logs:CreateLogGroup"
-        ],
-        "Resource" : "*"
-      },
-      {
-        "Effect" : "Allow",
-        "Action" : [
-          "sqs:*"
-        ],
-        "Resource" : aws_sqs_queue.event_sqs_queue.arn
-      }
-    ]
-  })
-
+  policy     = data.aws_iam_policy_document.policy.json
   depends_on = [resource.aws_lambda_function.handler_lambda, resource.aws_sqs_queue.event_sqs_queue]
```

Comment:
I believe this manual `depends_on` line isn't necessary now that we're using the `aws_iam_policy_document` resource. Feedback only, not a requested change.